algorithms/src/crypto_hash/blake2xs.rs

```diff
@@ -216,119 +129,4 @@ mod tests {
-                    padded_salt[i] = salt[i];
-                }
-                p[4] = u32::from_le_bytes(padded_salt[0 .. length/2].try_into().unwrap());
-                p[5] = u32::from_le_bytes(padded_salt[length/2 .. padded_salt.len()].try_into().unwrap());
-            } else {
-                p[4] = u32::from_le_bytes(salt[0 .. salt.len()/2].try_into().unwrap());
-                p[5] = u32::from_le_bytes(salt[salt.len()/2 .. salt.len()].try_into().unwrap());
-            }
-
-            // persona is also two words long
-            if persona.len() < length {
-                let mut padded_persona = GenericArray::<u8, <U32 as Div<U4>>::Output>::default();
-                for i in 0..persona.len() {
-                    padded_persona[i] = persona[i];
-                }
-                p[6] = u32::from_le_bytes(padded_persona[0 .. length/2].try_into().unwrap());
-                p[7] = u32::from_le_bytes(padded_persona[length/2 .. padded_persona.len()].try_into().unwrap());
-            } else {
-                p[6] = u32::from_le_bytes(persona[0 .. length/2].try_into().unwrap());
-                p[7] = u32::from_le_bytes(persona[length/2 .. persona.len()].try_into().unwrap());
-            }
-            p
-        }
-
-        // Initialize a random number generator.
-        let rng = &mut ChaChaRng::seed_from_u64(123456789u64);
-        
-        // Initialize the reference salt and persona.
-        const REFERENCE_SALT: [u8; 8] = [0u8; 8];
-        const REFERENCE_PERSONA: [u8; 8] = 0u64.to_le_bytes();
-        
-        // Run evaluations and enforce equality.
-        for _ in 0..ITERATIONS {
-            
-            // Initialize a reference implementation of Blake2s.
-            let reference_a = Blake2s::with_params(&[], &REFERENCE_SALT, &REFERENCE_PERSONA);
-
-            // Initialize a reference implementation of VarBlake2s.
-            let reference_b = VarBlake2s::with_params(&[], &REFERENCE_SALT, &REFERENCE_PERSONA, U32::to_usize());
-
-            // Initialize a reference implementation of VarBlake2s from a parameter block.
-            let reference_c = VarBlake2s::with_parameter_block(&u32_params_to_parameter_block(&[], &REFERENCE_SALT, &REFERENCE_PERSONA));
-
-            // Initialize a candidate implementation of Blake2s from the parameter block.
-            let candidate = VarBlake2s::with_parameter_block(&Blake2Xs::blake2s_parameter_block(0u64));
-            
-            // Sample a new input.
-            let input: [u8; 32] = rng.gen();
-            
-            // Compare the evaluation of the implementations.
-            assert_eq!(evaluate_blake2s(reference_a, input), evaluate_varblake2s(reference_b.clone(), input));
-            assert_eq!(evaluate_varblake2s(reference_b, input), evaluate_varblake2s(reference_c.clone(), input));
-            assert_eq!(evaluate_varblake2s(reference_c, input), evaluate_varblake2s(candidate, input));
-        }
-    }
```

Comment:
Keep the tests and implementation. They are used to check the test vectors match any updates to blake2s.

Reply:
If there is a mismatch, this allows us to debug what parameter changed

Reply:
I'll reintroduce the `blake2s` impl and test just to make sure those vectors is still correct. Note that this isn't the implementation that's currently used elsewhere in the codebase, which is why I removed it. We use the default parameters (no `persona`) and it is a plain `Blake2s256::digest` (unless we want to unify that under our own wrapper implementation?). 

Regarding the parameter block test: the `blake2` crate no longer supports specifying the parameter block, this was a public but hidden part of the API in `v0.9` but has been removed in `v0.10`. The simd implementation that we use for `Blake2Xs` also doesn't support a parameter block as input, the `Params` constructor must be used instead. I may have missed something but it seems that there's no parameter block left to test other than what is abstracted by the implementation itself and that's the dependency's concern?